src/service/promql/functions/histogram.rs

```diff
@@ -84,6 +87,7 @@ pub(crate) fn histogram_quantile(sample_time: i64, phi: f64, data: Value) -> Res
 
     let values = metrics_with_buckets
         .into_values()
+        .filter(|bucket| !bucket.buckets.is_empty())
```

Comment:
Cross verified with the promql code.

![image](https://github.com/openobserve/openobserve/assets/1695056/f6de2f69-1912-42fd-a38d-4dab711d41a0)